src/libs/actions/User.js

```diff
@@ -103,11 +112,46 @@ function setSecondaryLogin(login, password) {
     });
 }
 
+/**
+ * Validates a login given an accountID and validation code
+ *
+ * @param {Number} accountID
+ * @param {String} validateCode
+ * @returns {Promise}
+ *
+ */
+function validateLogin(accountID, validateCode) {
+    Onyx.merge(ONYXKEYS.ACCOUNT, {error: '', loading: true});
+
+    return API.ValidateEmail({
+        accountID,
+        validateCode,
+    }).then((response) => {
+        if (response.jsonCode === 200) {
+            const {authToken, email} = response;
+            if (!sessionAuthToken) {
+                createTemporaryLogin(authToken, email);
+            } else {
+                fetch();
+                Navigation.navigate(ROUTES.SETTINGS_PROFILE);
```

Comment:
Just clarifying the intention here. We do want to navigate immediately to the settings after the login is validated?

Reply:
We don't have to, I just figured it would make sense because it would show that the secondary login now exists in a validated state in that view.

Reply:
Ah Ok I think that works unless someone said otherwise. Was just curious since it seems like this would also happen if you were validating your primary login (and not secondary).

But I might be confused here and perhaps it's not possible for an e.cash user to ever get a "magic sign" in link.